# Graham Parcel Data

In [1]:
import geopandas as gpd
#reading in the file

par = gpd.read_file("../00_source_data/graham-parcels")
par

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  \
0        563000030016      075   Graham     None  1008555        0.0   
1      55380001006807      075   Graham     None  1008555        0.0   
2        557901000035      075   Graham     None  1008555        0.0   
3      55980000000309      075   Graham     None  1008555   150420.0   
4       5538000040980      075   Graham     None  1008555    25330.0   
...               ...      ...      ...      ...      ...        ...   
10315            None      075   Graham     None  1008555        0.0   
10316    564300009920      075   Graham     None  1008555   150790.0   
10317   5579000021400      075   Graham     None  1008555        0.0   
10318   5661040017690      075   Graham     None  1008555        0.0   
10319  56900000170004      075   Graham     None  1008555        0.0   

        LANDVAL                              LEGDECFULL MADDPREF MADDRNO  ...  \
0       39150.0                          FR SAM STEWART     None    None  ...   
1       45000.0                                    LT 7     None    None  ...   
2       10970.0  LOT 14 & UNNUMBERED LOT  BRIARWOOD HIL     None    None  ...   
3       30000.0                     LOT BEAR CREEK LEAD     None    None  ...   
4       49420.0                 3.01 AC LITTLE SNOWBIRD     None    None  ...   
...         ...                                     ...      ...     ...  ...   
10315       0.0                                    None     None    None  ...   
10316   83990.0                    FR ALFRED HOLLIFIELD     None    None  ...   
10317  235540.0                 FR LLOYD BERNIECE ADAMS     None    None  ...   
10318    7020.0          L4 &4A FR FRANK HOWELL   HEIRS     None    None  ...   
10319  108320.0  47.28 AC BEECH CREEK / FR- R.D JENKINS     None    None  ...   

      STRUCT STRUCTNO STRUCTYEAR SUBDIVISIO SUBOWNTYPE SUBSURFOWN SUNIT  SZIP  \
0          N     None       None       None       None       None  None  None   
1          N     None       None       None       None       None  None  None   
2          N     None       None       None       None       None  None  None   
3          Y     None       None       None       None       None  None  None   
4          Y     None       None       None       None       None  None  None   
...      ...      ...        ...        ...        ...        ...   ...   ...   
10315      N     None       None       None       None       None  None  None   
10316      Y     None       None       None       None       None  None  None   
10317      N     None       None       None       None       None  None  None   
10318      N     None       None       None       None       None  None  None   
10319      N     None       None       None       None       None  None  None   

       TRANSFDATE                                           geometry  
0      2022-07-21  POLYGON ((538626.906 603796.692, 538610.292 60...  
1      2022-07-21  POLYGON ((537075.448 586391.991, 537042.545 58...  
2      2022-07-21  POLYGON ((581671.771 593530.401, 581143.021 59...  
3      2022-07-21  POLYGON ((592549.910 588323.058, 592552.798 58...  
4      2022-07-21  POLYGON ((522641.646 575917.619, 522510.156 57...  
...           ...                                                ...  
10315  2022-07-21  POLYGON ((571048.198 634611.151, 571002.083 63...  
10316  2022-07-21  POLYGON ((572198.104 637588.328, 572295.677 63...  
10317  2022-07-21  POLYGON ((571485.539 594088.881, 571485.219 59...  
10318  2022-07-21  POLYGON ((567572.573 610849.661, 567507.927 61...  
10319  2022-07-21  POLYGON ((585136.239 611275.661, 585045.476 61...  

[10320 rows x 69 columns]

### Exploring the columns in the data frame

In [2]:
for col in par.columns:
    print(col)

ALTPARNO
CNTYFIPS
CNTYNAME
GISACRES
GNISID
IMPROVVAL
LANDVAL
LEGDECFULL
MADDPREF
MADDRNO
MADDSTNAME
MADDSTR
MADDSTSUF
MADDSTTYP
MAILADD
MAPREF
MCITY
MSTATE
MULTISTRUC
MUNIT
MZIP
NPARNO
OwnerName2
OWNFRST
OWNLAST
OWNNAME
OWNNAME2
OWNTYPE
PARNO
PARUSECD2
PARUSECODE
PARUSEDESC
PARUSEDSC2
PARVAL
PARVALTYPE
PRESENTVAL
RECAREANO
RECAREATX
REVDATETX
REVISEDATE
REVISEYEAR
SADDNO
SADDPREF
SADDSTNAME
SADDSTR
SADDSTSUF
SADDSTTYP
SALEDATE
SALEDATETX
SCITY
SITEADD
SOURCEAGNT
SOURCEDATE
SOURCEDATX
SOURCEREF
SSTATE
STCNTYFIPS
STFIPS
STNAME
STRUCT
STRUCTNO
STRUCTYEAR
SUBDIVISIO
SUBOWNTYPE
SUBSURFOWN
SUNIT
SZIP
TRANSFDATE
geometry


In [3]:
# checking how many NAs are in each column
for col in par.columns:
    # printing the total number of NAs in each column
    na_sum = par[col].isna().sum()
    # printing number of NAs if not 0
    if na_sum != 0:
        print(col, ":", par[col].isna().sum())

ALTPARNO : 180
GISACRES : 10320
LEGDECFULL : 633
MADDPREF : 10320
MADDRNO : 10320
MADDSTNAME : 468
MADDSTR : 10320
MADDSTSUF : 10320
MADDSTTYP : 10320
MAILADD : 451
MCITY : 436
MSTATE : 447
MUNIT : 10320
MZIP : 10320
OwnerName2 : 10320
OWNFRST : 10320
OWNLAST : 10320
OWNNAME : 436
OWNNAME2 : 5925
OWNTYPE : 10320
PARNO : 160
PARUSECD2 : 10320
PARUSECODE : 436
PARUSEDESC : 448
PARUSEDSC2 : 10320
RECAREATX : 10320
REVDATETX : 10320
REVISEYEAR : 10320
SADDNO : 4593
SADDPREF : 10315
SADDSTNAME : 569
SADDSTR : 569
SADDSTSUF : 10314
SADDSTTYP : 6104
SALEDATE : 10320
SALEDATETX : 10320
SCITY : 10320
SITEADD : 568
SOURCEDATE : 10320
SOURCEDATX : 10320
STRUCTNO : 10320
STRUCTYEAR : 10320
SUBDIVISIO : 10320
SUBOWNTYPE : 10320
SUBSURFOWN : 10320
SUNIT : 10318
SZIP : 10320


All columns with 10,320 NAs mean that the whole column is None.

In [4]:
#looking at values in columns of interest
print(par["PARUSEDESC"].unique())

['Rural homesite' 'Single family high value' 'Single family residential'
 'Manufactured home' 'Gymnasium' None 'Service garage' 'Commerical'
 'Public schools' 'Federal building' 'Churches' 'Garden apartment'
 'New parcels' 'Townhouse apartment' 'Light manufacturing'
 'Convenience store' 'Utilities' 'County building'
 'Service/repair building' 'Clubs/lodges/halls' 'Restaurant'
 'Municipal building' 'Funeral home/mortuary' 'State building'
 'Mini warehouse' 'Hotel/motel 3 stories or more' 'Office'
 'Prefab warehouse' 'Office/warehouse' 'Patio home' 'Warehouse'
 'Duplex/triplex' 'Shopping center strip' 'Granite shed'
 'Retail warehouse' 'Medical office' 'Shop/garage/auto repair'
 'Homes for the aged' 'Single family exceptional' 'Private school/college'
 'Institutional' 'Service station' 'Car wash' 'Bank'
 'Hotel/motel high rise' 'Fast food restaurant' 'Daycare center'
 'Supermarket']


In [5]:
#looking at values in columns of interest
print(par["PARUSEDSC2"].unique())

[None]


In [6]:
#looking at values in columns of interest
print(par["PARUSECD2"].unique())

[None]


In [7]:
#looking at values in columns of interest
print(par["STRUCTNO"].unique())

[None]


In [8]:
#looking at values in columns of interest
print(par["PRESENTVAL"].unique())

['N' 'Y']


Based off of all of the columns "PARUSEDESC" seems to be the most helpful. It lists the type of parcel the land is, so we could subset it to only residential parcels. Once that is subsetted we could possible multiple the housing count total for parcels that would contain more than one household on it. There are 448 missing values for this column which we should be aware of.

### Subsetting for only residential parcels
Questionable parcel types: 'New parcels', 'Light manufacturing', 'Granite shed', 'Private school/college', 'Institutional'

Columns to consider multiplying (other than basic residential types)
- 'Homes for the aged': this could be up to 100 different people living here. Would this be 100 different households?
- 'Duplex/triplex': this would be 2/3 different households?
- 'Private school/college': would each student living on campus be considered a different household? (if included)

In [9]:
#making a list of reisdential types
res =['Rural homesite','Single family high value', 'Single family residential',
 'Manufactured home', 'Townhouse apartment','Patio home', 'Duplex/triplex',
 'Single family exceptional', 'Garden apartment']

#subsetting df to only have residential parcels
par_res = par[par["PARUSEDESC"].isin(res)]
par_res

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  \
0        563000030016      075   Graham     None  1008555        0.0   
1      55380001006807      075   Graham     None  1008555        0.0   
2        557901000035      075   Graham     None  1008555        0.0   
3      55980000000309      075   Graham     None  1008555   150420.0   
4       5538000040980      075   Graham     None  1008555    25330.0   
...               ...      ...      ...      ...      ...        ...   
10313  569301020008P4      075   Graham     None  1008555        0.0   
10314    562000080026      075   Graham     None  1008555    76380.0   
10316    564300009920      075   Graham     None  1008555   150790.0   
10317   5579000021400      075   Graham     None  1008555        0.0   
10318   5661040017690      075   Graham     None  1008555        0.0   

        LANDVAL                              LEGDECFULL MADDPREF MADDRNO  ...  \
0       39150.0                          FR SAM STEWART     None    None  ...   
1       45000.0                                    LT 7     None    None  ...   
2       10970.0  LOT 14 & UNNUMBERED LOT  BRIARWOOD HIL     None    None  ...   
3       30000.0                     LOT BEAR CREEK LEAD     None    None  ...   
4       49420.0                 3.01 AC LITTLE SNOWBIRD     None    None  ...   
...         ...                                     ...      ...     ...  ...   
10313   63230.0                         FR CHENNAULT BJ     None    None  ...   
10314   22880.0                            BRYSON ACRES     None    None  ...   
10316   83990.0                    FR ALFRED HOLLIFIELD     None    None  ...   
10317  235540.0                 FR LLOYD BERNIECE ADAMS     None    None  ...   
10318    7020.0          L4 &4A FR FRANK HOWELL   HEIRS     None    None  ...   

      STRUCT STRUCTNO STRUCTYEAR SUBDIVISIO SUBOWNTYPE SUBSURFOWN SUNIT  SZIP  \
0          N     None       None       None       None       None  None  None   
1          N     None       None       None       None       None  None  None   
2          N     None       None       None       None       None  None  None   
3          Y     None       None       None       None       None  None  None   
4          Y     None       None       None       None       None  None  None   
...      ...      ...        ...        ...        ...        ...   ...   ...   
10313      N     None       None       None       None       None  None  None   
10314      Y     None       None       None       None       None  None  None   
10316      Y     None       None       None       None       None  None  None   
10317      N     None       None       None       None       None  None  None   
10318      N     None       None       None       None       None  None  None   

       TRANSFDATE                                           geometry  
0      2022-07-21  POLYGON ((538626.906 603796.692, 538610.292 60...  
1      2022-07-21  POLYGON ((537075.448 586391.991, 537042.545 58...  
2      2022-07-21  POLYGON ((581671.771 593530.401, 581143.021 59...  
3      2022-07-21  POLYGON ((592549.910 588323.058, 592552.798 58...  
4      2022-07-21  POLYGON ((522641.646 575917.619, 522510.156 57...  
...           ...                                                ...  
10313  2022-07-21  POLYGON ((590461.842 635097.467, 590462.917 63...  
10314  2022-07-21  POLYGON ((536186.349 609711.665, 536057.457 60...  
10316  2022-07-21  POLYGON ((572198.104 637588.328, 572295.677 63...  
10317  2022-07-21  POLYGON ((571485.539 594088.881, 571485.219 59...  
10318  2022-07-21  POLYGON ((567572.573 610849.661, 567507.927 61...  

[9133 rows x 69 columns]

In [10]:
# subsetting to parcels with a structure on them (STRUCT == Y)
par_res_struct = par_res[par_res["STRUCT"] == "Y"]
par_res_struct

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
3      55980000000309      075   Graham     None  1008555   150420.0  30000.0   
4       5538000040980      075   Graham     None  1008555    25330.0  49420.0   
7        566200100004      075   Graham     None  1008555   103090.0  80000.0   
9      557900030032PT      075   Graham     None  1008555    38140.0  20000.0   
13      5690000014860      075   Graham     None  1008555    40500.0  20000.0   
...               ...      ...      ...      ...      ...        ...      ...   
10307  5660121B002801      075   Graham     None  1008555    58290.0  20000.0   
10309  56700000000808      075   Graham     None  1008555   171600.0  23250.0   
10312   5661000013670      075   Graham     None  1008555    85350.0  25130.0   
10314    562000080026      075   Graham     None  1008555    76380.0  22880.0   
10316    564300009920      075   Graham     None  1008555   150790.0  83990.0   

                                LEGDECFULL MADDPREF MADDRNO  ... STRUCT  \
3                      LOT BEAR CREEK LEAD     None    None  ...      Y   
4                  3.01 AC LITTLE SNOWBIRD     None    None  ...      Y   
7                      FR AYNSLEY CLIFFORD     None    None  ...      Y   
9                            FR CLYDE HYDE     None    None  ...      Y   
13     TK 2      FR LARRY & GARY GRINSTAFF     None    None  ...      Y   
...                                    ...      ...     ...  ...    ...   
10307                     PARCEL 7 FORD ST     None    None  ...      Y   
10309             FR MILDRED FARR PETERSON     None    None  ...      Y   
10312                     FR A.F. GHORMLEY     None    None  ...      Y   
10314                         BRYSON ACRES     None    None  ...      Y   
10316                 FR ALFRED HOLLIFIELD     None    None  ...      Y   

      STRUCTNO STRUCTYEAR SUBDIVISIO SUBOWNTYPE SUBSURFOWN SUNIT  SZIP  \
3         None       None       None       None       None  None  None   
4         None       None       None       None       None  None  None   
7         None       None       None       None       None  None  None   
9         None       None       None       None       None  None  None   
13        None       None       None       None       None  None  None   
...        ...        ...        ...        ...        ...   ...   ...   
10307     None       None       None       None       None  None  None   
10309     None       None       None       None       None  None  None   
10312     None       None       None       None       None  None  None   
10314     None       None       None       None       None  None  None   
10316     None       None       None       None       None  None  None   

       TRANSFDATE                                           geometry  
3      2022-07-21  POLYGON ((592549.910 588323.058, 592552.798 58...  
4      2022-07-21  POLYGON ((522641.646 575917.619, 522510.156 57...  
7      2022-07-21  POLYGON ((561366.282 622819.898, 561299.244 62...  
9      2022-07-21  POLYGON ((574684.719 598606.036, 574704.750 59...  
13     2022-07-21  POLYGON ((587403.948 609281.296, 587496.375 60...  
...           ...                                                ...  
10307  2022-07-21  POLYGON ((567454.626 607216.854, 567336.827 60...  
10309  2022-07-21  POLYGON ((576448.354 609202.970, 576696.504 60...  
10312  2022-07-21  POLYGON ((565624.729 608329.348, 565623.869 60...  
10314  2022-07-21  POLYGON ((536186.349 609711.665, 536057.457 60...  
10316  2022-07-21  POLYGON ((572198.104 637588.328, 572295.677 63...  

[4762 rows x 69 columns]

In [11]:
#seeing how much of each type
par_res_struct["PARUSEDESC"].value_counts()

Rural homesite               2615
Single family residential     929
Manufactured home             887
Single family high value      313
Garden apartment                8
Patio home                      4
Duplex/triplex                  3
Townhouse apartment             2
Single family exceptional       1
Name: PARUSEDESC, dtype: int64

In [12]:
# checking for dupliate rows
duplicate = par_res_struct[par_res_struct.duplicated()]
duplicate

Empty GeoDataFrame
Columns: [ALTPARNO, CNTYFIPS, CNTYNAME, GISACRES, GNISID, IMPROVVAL, LANDVAL, LEGDECFULL, MADDPREF, MADDRNO, MADDSTNAME, MADDSTR, MADDSTSUF, MADDSTTYP, MAILADD, MAPREF, MCITY, MSTATE, MULTISTRUC, MUNIT, MZIP, NPARNO, OwnerName2, OWNFRST, OWNLAST, OWNNAME, OWNNAME2, OWNTYPE, PARNO, PARUSECD2, PARUSECODE, PARUSEDESC, PARUSEDSC2, PARVAL, PARVALTYPE, PRESENTVAL, RECAREANO, RECAREATX, REVDATETX, REVISEDATE, REVISEYEAR, SADDNO, SADDPREF, SADDSTNAME, SADDSTR, SADDSTSUF, SADDSTTYP, SALEDATE, SALEDATETX, SCITY, SITEADD, SOURCEAGNT, SOURCEDATE, SOURCEDATX, SOURCEREF, SSTATE, STCNTYFIPS, STFIPS, STNAME, STRUCT, STRUCTNO, STRUCTYEAR, SUBDIVISIO, SUBOWNTYPE, SUBSURFOWN, SUNIT, SZIP, TRANSFDATE, geometry]
Index: []

[0 rows x 69 columns]

## Reading in Tract Polygon data
Frist the Graham parcel data will be merged with the tract polygon data, using the "geometry" column.

In [13]:
tr_polygons = gpd.read_file("../00_source_data/tract_polygons/tl_2020_37_tract.shp")
tr_polygons

STATEFP COUNTYFP TRACTCE        GEOID     NAME              NAMELSAD  \
0         37      141  920300  37141920300     9203     Census Tract 9203   
1         37      141  990100  37141990100     9901     Census Tract 9901   
2         37      071  031600  37071031600      316      Census Tract 316   
3         37      071  031800  37071031800      318      Census Tract 318   
4         37      155  961801  37155961801  9618.01  Census Tract 9618.01   
...      ...      ...     ...          ...      ...                   ...   
2667      37      119  001504  37119001504    15.04    Census Tract 15.04   
2668      37      119  001505  37119001505    15.05    Census Tract 15.05   
2669      37      119  001603  37119001603    16.03    Census Tract 16.03   
2670      37      057  061903  37057061903   619.03   Census Tract 619.03   
2671      37      057  061502  37057061502   615.02   Census Tract 615.02   

      MTFCC FUNCSTAT      ALAND     AWATER     INTPTLAT      INTPTLON  \
0     G5020        S  236115472     605423  +34.6713171  -078.0020072   
1     G5020        S          0  133298393  +34.3492469  -077.6031267   
2     G5020        S   22744042      66304  +35.2681382  -081.2849294   
3     G5020        S    6071060       8149  +35.2623339  -081.2292201   
4     G5020        S  103973789          0  +34.6244596  -079.2476138   
...     ...      ...        ...        ...          ...           ...   
2667  G5020        S    3519177      10623  +35.2440467  -080.7584093   
2668  G5020        S    5239455      16974  +35.2743691  -080.7565551   
2669  G5020        S    2487106      15427  +35.2327298  -080.7705875   
2670  G5020        S   32520125      38446  +35.7914698  -080.0930999   
2671  G5020        S    7875244          0  +35.7964753  -080.2403633   

                                               geometry  
0     POLYGON ((-78.15648 34.67909, -78.15458 34.680...  
1     POLYGON ((-77.71433 34.29117, -77.71406 34.291...  
2     POLYGON ((-81.32560 35.26363, -81.32556 35.263...  
3     POLYGON ((-81.25719 35.26613, -81.25585 35.266...  
4     POLYGON ((-79.31012 34.59978, -79.31002 34.599...  
...                                                 ...  
2667  POLYGON ((-80.78122 35.24749, -80.78120 35.247...  
2668  POLYGON ((-80.76909 35.26604, -80.76889 35.266...  
2669  POLYGON ((-80.77565 35.23679, -80.77547 35.236...  
2670  POLYGON ((-80.14902 35.79641, -80.14753 35.797...  
2671  POLYGON ((-80.26038 35.78594, -80.26036 35.786...  

[2672 rows x 13 columns]

In [14]:
# subsetting tract polygons to only Graham county
gr_tr_polygons= tr_polygons[tr_polygons["COUNTYFP"]=="075"]
gr_tr_polygons

STATEFP COUNTYFP TRACTCE        GEOID  NAME           NAMELSAD  MTFCC  \
433      37      075  920100  37075920100  9201  Census Tract 9201  G5020   
434      37      075  920200  37075920200  9202  Census Tract 9202  G5020   
435      37      075  920300  37075920300  9203  Census Tract 9203  G5020   

    FUNCSTAT      ALAND    AWATER     INTPTLAT      INTPTLON  \
433        S  338391041  17011413  +35.3959955  -083.7941810   
434        S  114839853   2458723  +35.3117331  -083.7276987   
435        S  302958135   5587640  +35.2702395  -083.9567917   

                                              geometry  
433  POLYGON ((-84.02350 35.41142, -84.02343 35.411...  
434  POLYGON ((-83.86032 35.35888, -83.86002 35.359...  
435  POLYGON ((-84.03873 35.34853, -84.03868 35.348...

In [15]:
# changing the projection of the parcel data
par_proj = par_res_struct.to_crs(4269)
par_proj.head()

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
3   55980000000309      075   Graham     None  1008555   150420.0  30000.0   
4    5538000040980      075   Graham     None  1008555    25330.0  49420.0   
7     566200100004      075   Graham     None  1008555   103090.0  80000.0   
9   557900030032PT      075   Graham     None  1008555    38140.0  20000.0   
13   5690000014860      075   Graham     None  1008555    40500.0  20000.0   

                             LEGDECFULL MADDPREF MADDRNO  ... STRUCT STRUCTNO  \
3                   LOT BEAR CREEK LEAD     None    None  ...      Y     None   
4               3.01 AC LITTLE SNOWBIRD     None    None  ...      Y     None   
7                   FR AYNSLEY CLIFFORD     None    None  ...      Y     None   
9                         FR CLYDE HYDE     None    None  ...      Y     None   
13  TK 2      FR LARRY & GARY GRINSTAFF     None    None  ...      Y     None   

   STRUCTYEAR SUBDIVISIO SUBOWNTYPE SUBSURFOWN SUNIT  SZIP  TRANSFDATE  \
3        None       None       None       None  None  None  2022-07-21   
4        None       None       None       None  None  None  2022-07-21   
7        None       None       None       None  None  None  2022-07-21   
9        None       None       None       None  None  None  2022-07-21   
13       None       None       None       None  None  None  2022-07-21   

                                             geometry  
3   POLYGON ((-83.71750 35.27462, -83.71747 35.274...  
4   POLYGON ((-83.94936 35.23123, -83.94980 35.231...  
7   POLYGON ((-83.82747 35.36518, -83.82769 35.364...  
9   POLYGON ((-83.77894 35.30050, -83.77886 35.300...  
13  POLYGON ((-83.73807 35.33147, -83.73773 35.331...  

[5 rows x 69 columns]

In [16]:
# merging the Graham parcel data with the tract polygon data
graham_par_with_tract= par_proj.sjoin(gr_tr_polygons, how="left", op="within")
graham_par_with_tract

/Users/rachelrichards/opt/miniconda3/envs/geo/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3444: FutureWarning: The `op` parameter is deprecated and will be removed in a future release. Please use the `predicate` parameter instead.
  exec(code_obj, self.user_global_ns, self.user_ns)


ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
3      55980000000309      075   Graham     None  1008555   150420.0  30000.0   
4       5538000040980      075   Graham     None  1008555    25330.0  49420.0   
7        566200100004      075   Graham     None  1008555   103090.0  80000.0   
9      557900030032PT      075   Graham     None  1008555    38140.0  20000.0   
13      5690000014860      075   Graham     None  1008555    40500.0  20000.0   
...               ...      ...      ...      ...      ...        ...      ...   
10307  5660121B002801      075   Graham     None  1008555    58290.0  20000.0   
10309  56700000000808      075   Graham     None  1008555   171600.0  23250.0   
10312   5661000013670      075   Graham     None  1008555    85350.0  25130.0   
10314    562000080026      075   Graham     None  1008555    76380.0  22880.0   
10316    564300009920      075   Graham     None  1008555   150790.0  83990.0   

                                LEGDECFULL MADDPREF MADDRNO  ... TRACTCE  \
3                      LOT BEAR CREEK LEAD     None    None  ...  920200   
4                  3.01 AC LITTLE SNOWBIRD     None    None  ...  920300   
7                      FR AYNSLEY CLIFFORD     None    None  ...  920100   
9                            FR CLYDE HYDE     None    None  ...  920200   
13     TK 2      FR LARRY & GARY GRINSTAFF     None    None  ...  920200   
...                                    ...      ...     ...  ...     ...   
10307                     PARCEL 7 FORD ST     None    None  ...  920300   
10309             FR MILDRED FARR PETERSON     None    None  ...  920200   
10312                     FR A.F. GHORMLEY     None    None  ...  920300   
10314                         BRYSON ACRES     None    None  ...  920300   
10316                 FR ALFRED HOLLIFIELD     None    None  ...  920100   

             GEOID  NAME           NAMELSAD  MTFCC FUNCSTAT        ALAND  \
3      37075920200  9202  Census Tract 9202  G5020        S  114839853.0   
4      37075920300  9203  Census Tract 9203  G5020        S  302958135.0   
7      37075920100  9201  Census Tract 9201  G5020        S  338391041.0   
9      37075920200  9202  Census Tract 9202  G5020        S  114839853.0   
13     37075920200  9202  Census Tract 9202  G5020        S  114839853.0   
...            ...   ...                ...    ...      ...          ...   
10307  37075920300  9203  Census Tract 9203  G5020        S  302958135.0   
10309  37075920200  9202  Census Tract 9202  G5020        S  114839853.0   
10312  37075920300  9203  Census Tract 9203  G5020        S  302958135.0   
10314  37075920300  9203  Census Tract 9203  G5020        S  302958135.0   
10316  37075920100  9201  Census Tract 9201  G5020        S  338391041.0   

           AWATER     INTPTLAT      INTPTLON  
3       2458723.0  +35.3117331  -083.7276987  
4       5587640.0  +35.2702395  -083.9567917  
7      17011413.0  +35.3959955  -083.7941810  
9       2458723.0  +35.3117331  -083.7276987  
13      2458723.0  +35.3117331  -083.7276987  
...           ...          ...           ...  
10307   5587640.0  +35.2702395  -083.9567917  
10309   2458723.0  +35.3117331  -083.7276987  
10312   5587640.0  +35.2702395  -083.9567917  
10314   5587640.0  +35.2702395  -083.9567917  
10316  17011413.0  +35.3959955  -083.7941810  

[4762 rows x 82 columns]

In [17]:
#checking the full FIPS code in merged df
graham_par_with_tract["GEOID"].unique()

array(['37075920200', '37075920300', '37075920100', nan], dtype=object)

In [18]:
# seeing which units have nan for FIPs code
na_check=graham_par_with_tract[graham_par_with_tract["GEOID"].isna()]
na_check

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  \
399      557900030021      075   Graham     None  1008555    39680.0   
468      566007040010      075   Graham     None  1008555   121630.0   
1406     566016200100      075   Graham     None  1008555   338280.0   
1764     650903000001      075   Graham     None  1008555   206450.0   
2074   56620008001201      075   Graham     None  1008555    12160.0   
2125     553800010103      075   Graham     None  1008555     7740.0   
2236     566200010009      075   Graham     None  1008555   382830.0   
3369     566200080005      075   Graham     None  1008555   371730.0   
4043    5579000035090      075   Graham     None  1008555    17180.0   
4053    5661040016490      075   Graham     None  1008555     5970.0   
4569     553800010017      075   Graham     None  1008555    80680.0   
4927     556902001190      075   Graham     None  1008555    18490.0   
5205     566008030027      075   Graham     None  1008555    57130.0   
5325     566000000033      075   Graham     None  1008555    66940.0   
5527     554900030006      075   Graham     None  1008555    43050.0   
5700     554900030006      075   Graham     None  1008555    43050.0   
5778     557900008560      075   Graham     None  1008555    65160.0   
5887    5588000040860      075   Graham     None  1008555   522950.0   
6256    5588000020320      075   Graham     None  1008555    38840.0   
6300     650903000001      075   Graham     None  1008555   206450.0   
6475     553800010031      075   Graham     None  1008555    23130.0   
6608   55380001010201      075   Graham     None  1008555   229620.0   
7149     650900000001      075   Graham     None  1008555    31700.0   
7197     650903000001      075   Graham     None  1008555   206450.0   
7393     553800010099      075   Graham     None  1008555    32520.0   
7430     553800010017      075   Graham     None  1008555    80680.0   
8124     566200080012      075   Graham     None  1008555    44440.0   
8350     566200080011      075   Graham     None  1008555    25050.0   
8421     566200080001      075   Graham     None  1008555    28390.0   
9370     566016200003      075   Graham     None  1008555    44950.0   
9466     566008030028      075   Graham     None  1008555    48460.0   
9547     650900000025      075   Graham     None  1008555    74860.0   
9686     566000000035      075   Graham     None  1008555   176880.0   
9805     650904000008      075   Graham     None  1008555    12140.0   
9842     556902001020      075   Graham     None  1008555    68810.0   
10071    650903000007      075   Graham     None  1008555    63950.0   

         LANDVAL                              LEGDECFULL MADDPREF MADDRNO  \
399      43650.0       FR PHILLIPS & JORDAN &   GHORMLEY     None    None   
468      48540.0               FROM HILDRED MILLSAPS L/T     None    None   
1406     71140.0                    FR PHILLIPS & JORDAN     None    None   
1764     60000.0                LT 1 S 3 NANTAHALA ACRES     None    None   
2074     20000.0                     FR RICHARD WALDROUP     None    None   
2125    250980.0             BEMIS TK103 FR-DALTON ADAMS     None    None   
2236    121590.0       L8 9 10;PT 11 S1 ORR&ORR +13.95AC     None    None   
3369    394150.0                TK 2  FR PAULINE JENKINS     None    None   
4043     47910.0  LOT #17 18 19            W H HYDE FARM     None    None   
4053      3600.0                      FR  JOE B GHORMLEY     None    None   
4569   3656600.0                 TK 1 COMBINED 3 PARCELS     None    None   
4927     20330.0                 FR HUGH DARRELL ROBERTS     None    None   
5205     19400.0                      L27 BLK3 FR JORDAN     None    None   
5325    202150.0                         BK-4  HOMEPLACE     None    None   
5527     65870.0                       FR MARTIN DOCKERY     None    None   
5700     65870.0                       FR MARTIN DOCKERY     None    None   
5778    121100.0                       

## Reading in Tract Demographic data
The Graham parcel and tract polygon data will be merged with the tract demographic data, using the full FIPS code.

In [19]:
import geopandas as gpd
#reading in the file

tr_dem = gpd.read_file("../00_source_data/2020_counts_NC_tract.csv")
tr_dem

county_and_tra pop_count total_housing  \
0        Census Tract 201, Alamance County    4040.0        2059.0   
1        Census Tract 202, Alamance County    4137.0        1786.0   
2     Census Tract 203.01, Alamance County    4213.0        1677.0   
3     Census Tract 203.02, Alamance County    3974.0        1435.0   
4        Census Tract 204, Alamance County    6556.0        2710.0   
...                                    ...       ...           ...   
2667      Census Tract 535.17, Wake County    4663.0        1765.0   
2668      Census Tract 535.18, Wake County    3528.0        1590.0   
2669      Census Tract 535.19, Wake County    3954.0        2014.0   
2670      Census Tract 535.20, Wake County    5857.0        2630.0   
2671      Census Tract 535.21, Wake County    3638.0        1397.0   

     occ_housing unocc_housing total_group total_inst adult_jail  \
0         1894.0         165.0        43.0        3.0        0.0   
1         1607.0         179.0        57.0        0.0        0.0   
2         1571.0         106.0       160.0      133.0        0.0   
3         1360.0          75.0         7.0        7.0        0.0   
4         2464.0         246.0        32.0       12.0        0.0   
...          ...           ...         ...        ...        ...   
2667      1676.0          89.0        25.0        0.0        0.0   
2668      1533.0          57.0        22.0        6.0        0.0   
2669      1848.0         166.0         0.0        0.0        0.0   
2670      2506.0         124.0         0.0        0.0        0.0   
2671      1351.0          46.0         3.0        1.0        0.0   

     juvenille_jail nursing_homes other_inst non_inst_total college_housing  \
0               3.0           0.0        0.0           40.0             0.0   
1               0.0           0.0        0.0           57.0             0.0   
2               0.0         133.0        0.0           27.0             0.0   
3               7.0           0.0        0.0            0.0             0.0   
4               0.0          12.0        0.0           20.0             0.0   
...             ...           ...        ...            ...             ...   
2667            0.0           0.0        0.0           25.0             0.0   
2668            6.0           0.0        0.0           16.0             0.0   
2669            0.0           0.0        0.0            0.0             0.0   
2670            0.0           0.0        0.0            0.0             0.0   
2671            1.0           0.0        0.0            2.0             0.0   

     military_housing other_non_inst state_fips county_fips tract_fips  \
0                 0.0           40.0         37         001     020100   
1                 0.0           57.0         37         001     020200   
2                 0.0           27.0         37         001     020301   
3                 0.0            0.0         37         001     020302   
4                 0.0           20.0         37         001     020400   
...               ...            ...        ...         ...        ...   
2667              0.0           25.0         37         183     053517   
2668              0.0           16.0         37         183     053518   
2669              0.0            0.0         37         183     053519   
2670              0.0            0.0         37         183     053520   
2671              0.0            2.0         37         183     053521   

        full_fips geometry  
0     37001020100     None  
1     37001020200     None  
2     37001020301     None  
3     37001020302     None  
4     37001020400     None  
...           ...      ...  
2667  37183053517     None  
2668  37183053518     None  
2669  37183053519     None  
2670  37183053520     None  
2671  37183053521     None  

[2672 rows x 20 columns]

In [20]:
# subsetting for only Graham county
gr_tr_dem= tr_dem[tr_dem["county_fips"]== "075"]
gr_tr_dem

county_and_tra pop_count total_housing occ_housing  \
2060  Census Tract 9201, Graham County    2019.0        1903.0       931.0   
2061  Census Tract 9202, Graham County    2703.0        1414.0      1058.0   
2062  Census Tract 9203, Graham County    3308.0        1895.0      1328.0   

     unocc_housing total_group total_inst adult_jail juvenille_jail  \
2060         972.0         0.0        0.0        0.0            0.0   
2061         356.0       179.0        0.0        0.0            0.0   
2062         567.0        96.0       96.0       23.0            0.0   

     nursing_homes other_inst non_inst_total college_housing military_housing  \
2060           0.0        0.0            0.0             0.0              0.0   
2061           0.0        0.0          179.0             0.0              0.0   
2062          73.0        0.0            0.0             0.0              0.0   

     other_non_inst state_fips county_fips tract_fips    full_fips geometry  
2060            0.0         37         075     920100  37075920100     None  
2061          179.0         37         075     920200  37075920200     None  
2062            0.0         37         075     920300  37075920300     None

In [21]:
# checking full FIPs codes
gr_tr_dem["full_fips"].unique()

array(['37075920100', '37075920200', '37075920300'], dtype=object)

In [22]:
# merging tract demographic data with parcel data
graham_final = graham_par_with_tract.merge(gr_tr_dem, left_on="GEOID", right_on = "full_fips")
graham_final

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
0     55980000000309      075   Graham     None  1008555   150420.0  30000.0   
1     557900030032PT      075   Graham     None  1008555    38140.0  20000.0   
2      5690000014860      075   Graham     None  1008555    40500.0  20000.0   
3       567002005740      075   Graham     None  1008555    97710.0  68200.0   
4       566200040004      075   Graham     None  1008555    65210.0  43260.0   
...              ...      ...      ...      ...      ...        ...      ...   
4721   6622000018060      075   Graham     None  1008555    25540.0  30000.0   
4722   6613000014730      075   Graham     None  1008555    33810.0  20000.0   
4723    566200010013      075   Graham     None  1008555    55950.0  29100.0   
4724  568300030008PT      075   Graham     None  1008555   144160.0  22970.0   
4725    564300009920      075   Graham     None  1008555   150790.0  83990.0   

                               LEGDECFULL MADDPREF MADDRNO  ... other_inst  \
0                     LOT BEAR CREEK LEAD     None    None  ...        0.0   
1                           FR CLYDE HYDE     None    None  ...        0.0   
2     TK 2      FR LARRY & GARY GRINSTAFF     None    None  ...        0.0   
3                        FR FRANK SCHULER     None    None  ...        0.0   
4                                TK 3B &4     None    None  ...        0.0   
...                                   ...      ...     ...  ...        ...   
4721                           PANTHER CK     None    None  ...        0.0   
4722                       SHALL STAND RD     None    None  ...        0.0   
4723                 L13 S1 ORR & ORR SUB     None    None  ...        0.0   
4724                   FR-SCOTT WAYCASTER     None    None  ...        0.0   
4725                 FR ALFRED HOLLIFIELD     None    None  ...        0.0   

     non_inst_total college_housing military_housing other_non_inst  \
0             179.0             0.0              0.0          179.0   
1             179.0             0.0              0.0          179.0   
2             179.0             0.0              0.0          179.0   
3             179.0             0.0              0.0          179.0   
4             179.0             0.0              0.0          179.0   
...             ...             ...              ...            ...   
4721            0.0             0.0              0.0            0.0   
4722            0.0             0.0              0.0            0.0   
4723            0.0             0.0              0.0            0.0   
4724            0.0             0.0              0.0            0.0   
4725            0.0             0.0              0.0            0.0   

     state_fips county_fips tract_fips    full_fips geometry_y  
0            37         075     920200  37075920200       None  
1            37         075     920200  37075920200       None  
2            37         075     920200  37075920200       None  
3            37         075     920200  37075920200       None  
4            37         075     920200  37075920200       None  
...         ...         ...        ...          ...        ...  
4721         37         075     920100  37075920100       None  
4722         37         075     920100  37075920100       None  
4723         37         075     920100  37075920100       None  
4724         37         075     920100  37075920100       None  
4725         37         075     920100  37075920100       None  

[4726 rows x 102 columns]

# Generating estimates
I start by making sure all units in the same tract have the same total housing population. Then I create a new column for "estimated_housing_pop2" that contains the number of etimate housing units per row. This will be 1 for almost all of the columns except for duplex/triplex, where it will be 2. A second column will be created, "estimated_housing_pop2" where duplex/triplex will be counted as 3. I will then group by for both columns to get their total housing units per tract, for each counting method. Finally two more columns will be created which will find the difference between the estimates and the estimates I calculated.

In [23]:
sub_tract = graham_final[graham_final["full_fips"] == "37075920300"]
sub_tract

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
1361   5538000040980      075   Graham     None  1008555    25330.0  49420.0   
1362   5569000018430      075   Graham     None  1008555    34860.0  20000.0   
1363    5660111A0010      075   Graham     None  1008555    54470.0  20000.0   
1364   5660151E0057A      075   Graham     None  1008555    36620.0  20000.0   
1365  55790000798011      075   Graham     None  1008555   110530.0   4680.0   
...              ...      ...      ...      ...      ...        ...      ...   
2822   5661000019860      075   Graham     None  1008555    20410.0  42360.0   
2823   5660150018840      075   Graham     None  1008555    44240.0  20220.0   
2824  5660121B002801      075   Graham     None  1008555    58290.0  20000.0   
2825   5661000013670      075   Graham     None  1008555    85350.0  25130.0   
2826    562000080026      075   Graham     None  1008555    76380.0  22880.0   

                                LEGDECFULL MADDPREF MADDRNO  ... other_inst  \
1361               3.01 AC LITTLE SNOWBIRD     None    None  ...        0.0   
1362                         FR J S HOWELL     None    None  ...        0.0   
1363                  CHURCH ST LT10 BLK1A     None    None  ...        0.0   
1364  PT OF L57 BK1E  CIRCLE ST *2 PARCELS     None    None  ...        0.0   
1365    FR-GRAHAM COUNTY BOARD OF EDU LT#7     None    None  ...        0.0   
...                                    ...      ...     ...  ...        ...   
2822                      FR ELWOOD JORDAN     None    None  ...        0.0   
2823                      FR FRANK D JAMES     None    None  ...        0.0   
2824                      PARCEL 7 FORD ST     None    None  ...        0.0   
2825                      FR A.F. GHORMLEY     None    None  ...        0.0   
2826                          BRYSON ACRES     None    None  ...        0.0   

     non_inst_total college_housing military_housing other_non_inst  \
1361            0.0             0.0              0.0            0.0   
1362            0.0             0.0              0.0            0.0   
1363            0.0             0.0              0.0            0.0   
1364            0.0             0.0              0.0            0.0   
1365            0.0             0.0              0.0            0.0   
...             ...             ...              ...            ...   
2822            0.0             0.0              0.0            0.0   
2823            0.0             0.0              0.0            0.0   
2824            0.0             0.0              0.0            0.0   
2825            0.0             0.0              0.0            0.0   
2826            0.0             0.0              0.0            0.0   

     state_fips county_fips tract_fips    full_fips geometry_y  
1361         37         075     920300  37075920300       None  
1362         37         075     920300  37075920300       None  
1363         37         075     920300  37075920300       None  
1364         37         075     920300  37075920300       None  
1365         37         075     920300  37075920300       None  
...         ...         ...        ...          ...        ...  
2822         37         075     920300  37075920300       None  
2823         37         075     920300  37075920300       None  
2824         37         075     920300  37075920300       None  
2825         37         075     920300  37075920300       None  
2826         37         075     920300  37075920300       None  

[1466 rows x 102 columns]

In [24]:
sub_tract["total_housing"].unique()

array(['1895.0'], dtype=object)

In [25]:
graham_final["total_housing"].unique()

array(['1414.0', '1895.0', '1903.0'], dtype=object)

In [26]:
# checking the number of "Duplex/triplex" rows
duplex_check = graham_final[graham_final["PARUSEDESC"] == "Duplex/triplex"]
duplex_check

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
1696   5578000000841      075   Graham     None  1008555   109180.0  20000.0   
1844  56601204551130      075   Graham     None  1008555   279850.0  63000.0   
2202  56601204551130      075   Graham     None  1008555   279850.0  63000.0   

            LEGDECFULL MADDPREF MADDRNO  ... other_inst non_inst_total  \
1696  FR CORA ANDERSON     None    None  ...        0.0            0.0   
1844             MOTEL     None    None  ...        0.0            0.0   
2202             MOTEL     None    None  ...        0.0            0.0   

     college_housing military_housing other_non_inst state_fips county_fips  \
1696             0.0              0.0            0.0         37         075   
1844             0.0              0.0            0.0         37         075   
2202             0.0              0.0            0.0         37         075   

     tract_fips    full_fips geometry_y  
1696     920300  37075920300       None  
1844     920300  37075920300       None  
2202     920300  37075920300       None  

[3 rows x 102 columns]

In [27]:
# creating estimate columns with all values as 1
graham_final["estimated_housing_pop2"] = 1
graham_final["estimated_housing_pop3"] = 1

In [28]:
# generating the column "estimated_housing_pop2"
# if "PARUSEDESC" is "Duplex/triplex" then "estimated_housing_pop2" is 2
import numpy as np

graham_final["estimated_housing_pop2"]= np.where(graham_final["PARUSEDESC"] == "Duplex/triplex", 2, 1)
graham_final["estimated_housing_pop3"]= np.where(graham_final["PARUSEDESC"] == "Duplex/triplex", 3, 1)
graham_final

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
0     55980000000309      075   Graham     None  1008555   150420.0  30000.0   
1     557900030032PT      075   Graham     None  1008555    38140.0  20000.0   
2      5690000014860      075   Graham     None  1008555    40500.0  20000.0   
3       567002005740      075   Graham     None  1008555    97710.0  68200.0   
4       566200040004      075   Graham     None  1008555    65210.0  43260.0   
...              ...      ...      ...      ...      ...        ...      ...   
4721   6622000018060      075   Graham     None  1008555    25540.0  30000.0   
4722   6613000014730      075   Graham     None  1008555    33810.0  20000.0   
4723    566200010013      075   Graham     None  1008555    55950.0  29100.0   
4724  568300030008PT      075   Graham     None  1008555   144160.0  22970.0   
4725    564300009920      075   Graham     None  1008555   150790.0  83990.0   

                               LEGDECFULL MADDPREF MADDRNO  ...  \
0                     LOT BEAR CREEK LEAD     None    None  ...   
1                           FR CLYDE HYDE     None    None  ...   
2     TK 2      FR LARRY & GARY GRINSTAFF     None    None  ...   
3                        FR FRANK SCHULER     None    None  ...   
4                                TK 3B &4     None    None  ...   
...                                   ...      ...     ...  ...   
4721                           PANTHER CK     None    None  ...   
4722                       SHALL STAND RD     None    None  ...   
4723                 L13 S1 ORR & ORR SUB     None    None  ...   
4724                   FR-SCOTT WAYCASTER     None    None  ...   
4725                 FR ALFRED HOLLIFIELD     None    None  ...   

     college_housing military_housing other_non_inst state_fips county_fips  \
0                0.0              0.0          179.0         37         075   
1                0.0              0.0          179.0         37         075   
2                0.0              0.0          179.0         37         075   
3                0.0              0.0          179.0         37         075   
4                0.0              0.0          179.0         37         075   
...              ...              ...            ...        ...         ...   
4721             0.0              0.0            0.0         37         075   
4722             0.0              0.0            0.0         37         075   
4723             0.0              0.0            0.0         37         075   
4724             0.0              0.0            0.0         37         075   
4725             0.0              0.0            0.0         37         075   

     tract_fips    full_fips geometry_y estimated_housing_pop2  \
0        920200  37075920200       None                      1   
1        920200  37075920200       None                      1   
2        920200  37075920200       None                      1   
3        920200  37075920200       None                      1   
4        920200  37075920200       None                      1   
...         ...          ...        ...                    ...   
4721     920100  37075920100       None                      1   
4722     920100  37075920100       None                      1   
4723     920100  37075920100       None                      1   
4724     920100  37075920100       None                      1   
4725     920100  37075920100       None                      1   

     estimated_housing_pop3  
0                         1  
1                         1  
2                         1  
3                         1  
4                         1  
...                     ...  
4721                      1  
4722                      1  
4723                      1  
4724                      1  
4725                      1  

[4726 rows x 104 columns]

In [29]:
# checking the values in "estimated_housing_pop2"
graham_final["estimated_housing_pop2"].unique()

array([1, 2])

In [30]:
# checking the values in "estimated_housing_pop3"
graham_final["estimated_housing_pop3"].unique()

array([1, 3])

In [31]:
# checking the number of "Duplex/triplex" rows anf their values
duplex_check2 = graham_final[graham_final["PARUSEDESC"] == "Duplex/triplex"]
duplex_check2

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
1696   5578000000841      075   Graham     None  1008555   109180.0  20000.0   
1844  56601204551130      075   Graham     None  1008555   279850.0  63000.0   
2202  56601204551130      075   Graham     None  1008555   279850.0  63000.0   

            LEGDECFULL MADDPREF MADDRNO  ... college_housing military_housing  \
1696  FR CORA ANDERSON     None    None  ...             0.0              0.0   
1844             MOTEL     None    None  ...             0.0              0.0   
2202             MOTEL     None    None  ...             0.0              0.0   

     other_non_inst state_fips county_fips tract_fips    full_fips geometry_y  \
1696            0.0         37         075     920300  37075920300       None   
1844            0.0         37         075     920300  37075920300       None   
2202            0.0         37         075     920300  37075920300       None   

     estimated_housing_pop2 estimated_housing_pop3  
1696                      2                      3  
1844                      2                      3  
2202                      2                      3  

[3 rows x 104 columns]

### Groupby Tract
Now that the columns for the estimates are created I will groupy the tract to generate the total housing population estimates for each tract. The first groupby will be for the method where duplex/triplex is counted as 2 and the second is for the method where duplex/triplex is counted as 3.

In [32]:
# groupby full_fips for "estimated_housing_pop2" and "estimated_housing_pop3" sums per tract
graham_final['total_estimated_housing_pop2'] = graham_final.groupby("full_fips", sort=False)["estimated_housing_pop2"].transform('sum')
graham_final['total_estimated_housing_pop3'] = graham_final.groupby("full_fips", sort=False)["estimated_housing_pop3"].transform('sum')
graham_final.head()

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
0  55980000000309      075   Graham     None  1008555   150420.0  30000.0   
1  557900030032PT      075   Graham     None  1008555    38140.0  20000.0   
2   5690000014860      075   Graham     None  1008555    40500.0  20000.0   
3    567002005740      075   Graham     None  1008555    97710.0  68200.0   
4    566200040004      075   Graham     None  1008555    65210.0  43260.0   

                            LEGDECFULL MADDPREF MADDRNO  ... other_non_inst  \
0                  LOT BEAR CREEK LEAD     None    None  ...          179.0   
1                        FR CLYDE HYDE     None    None  ...          179.0   
2  TK 2      FR LARRY & GARY GRINSTAFF     None    None  ...          179.0   
3                     FR FRANK SCHULER     None    None  ...          179.0   
4                             TK 3B &4     None    None  ...          179.0   

  state_fips county_fips tract_fips    full_fips geometry_y  \
0         37         075     920200  37075920200       None   
1         37         075     920200  37075920200       None   
2         37         075     920200  37075920200       None   
3         37         075     920200  37075920200       None   
4         37         075     920200  37075920200       None   

  estimated_housing_pop2 estimated_housing_pop3 total_estimated_housing_pop2  \
0                      1                      1                         1361   
1                      1                      1                         1361   
2                      1                      1                         1361   
3                      1                      1                         1361   
4                      1                      1                         1361   

  total_estimated_housing_pop3  
0                         1361  
1                         1361  
2                         1361  
3                         1361  
4                         1361  

[5 rows x 106 columns]

### Percent Difference
Calculating the percent differnce between our estimates and the total housing population counts. There will be two different percent differences, "housing_per_diff2" and "housing_per_diff3". The first diff is for the method with counting duplex/triplex as 2, while the second diff is for counting them as 3.

In [33]:
# converting all populations and estimates to floats
graham_final["total_housing"]= graham_final["total_housing"].astype(float)
graham_final["total_estimated_housing_pop2"]= graham_final["total_estimated_housing_pop2"].astype(float)
graham_final["total_estimated_housing_pop3"]= graham_final["total_estimated_housing_pop3"].astype(float)

In [34]:
# creating columns for percent difference
graham_final["housing_per_diff2"] = ((graham_final["total_estimated_housing_pop2"] - graham_final["total_housing"])/graham_final["total_housing"])*100

graham_final["housing_per_diff3"] = ((graham_final["total_estimated_housing_pop3"] - graham_final["total_housing"])/graham_final["total_housing"])*100

graham_final.head()

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
0  55980000000309      075   Graham     None  1008555   150420.0  30000.0   
1  557900030032PT      075   Graham     None  1008555    38140.0  20000.0   
2   5690000014860      075   Graham     None  1008555    40500.0  20000.0   
3    567002005740      075   Graham     None  1008555    97710.0  68200.0   
4    566200040004      075   Graham     None  1008555    65210.0  43260.0   

                            LEGDECFULL MADDPREF MADDRNO  ... county_fips  \
0                  LOT BEAR CREEK LEAD     None    None  ...         075   
1                        FR CLYDE HYDE     None    None  ...         075   
2  TK 2      FR LARRY & GARY GRINSTAFF     None    None  ...         075   
3                     FR FRANK SCHULER     None    None  ...         075   
4                             TK 3B &4     None    None  ...         075   

  tract_fips    full_fips geometry_y estimated_housing_pop2  \
0     920200  37075920200       None                      1   
1     920200  37075920200       None                      1   
2     920200  37075920200       None                      1   
3     920200  37075920200       None                      1   
4     920200  37075920200       None                      1   

  estimated_housing_pop3 total_estimated_housing_pop2  \
0                      1                       1361.0   
1                      1                       1361.0   
2                      1                       1361.0   
3                      1                       1361.0   
4                      1                       1361.0   

  total_estimated_housing_pop3 housing_per_diff2 housing_per_diff3  
0                       1361.0         -3.748232         -3.748232  
1                       1361.0         -3.748232         -3.748232  
2                       1361.0         -3.748232         -3.748232  
3                       1361.0         -3.748232         -3.748232  
4                       1361.0         -3.748232         -3.748232  

[5 rows x 108 columns]

In [35]:
# seeing percent diff per tract by removing duplicates
per_diff_check = graham_final.drop_duplicates(subset=['full_fips'], keep='first')
per_diff_check

ALTPARNO CNTYFIPS CNTYNAME GISACRES   GNISID  IMPROVVAL  LANDVAL  \
0     55980000000309      075   Graham     None  1008555   150420.0  30000.0   
1361   5538000040980      075   Graham     None  1008555    25330.0  49420.0   
2827    566200100004      075   Graham     None  1008555   103090.0  80000.0   

                   LEGDECFULL MADDPREF MADDRNO  ... county_fips tract_fips  \
0         LOT BEAR CREEK LEAD     None    None  ...         075     920200   
1361  3.01 AC LITTLE SNOWBIRD     None    None  ...         075     920300   
2827      FR AYNSLEY CLIFFORD     None    None  ...         075     920100   

        full_fips geometry_y estimated_housing_pop2 estimated_housing_pop3  \
0     37075920200       None                      1                      1   
1361  37075920300       None                      1                      1   
2827  37075920100       None                      1                      1   

     total_estimated_housing_pop2 total_estimated_housing_pop3  \
0                          1361.0                       1361.0   
1361                       1469.0                       1472.0   
2827                       1899.0                       1899.0   

     housing_per_diff2 housing_per_diff3  
0            -3.748232         -3.748232  
1361        -22.480211        -22.321900  
2827         -0.210194         -0.210194  

[3 rows x 108 columns]

In [36]:
#grouping by tract

import pandas as pd
perc_diff_df= pd.DataFrame(graham_final[["full_fips","total_housing", "total_estimated_housing_pop3", "housing_per_diff2", "housing_per_diff3"]])

# seeing percent diff per tract by removing duplicates
per_diff_check = perc_diff_df.drop_duplicates(subset=['full_fips'], keep='first')
per_diff_check

full_fips  total_housing  total_estimated_housing_pop3  \
0     37075920200         1414.0                        1361.0   
1361  37075920300         1895.0                        1472.0   
2827  37075920100         1903.0                        1899.0   

      housing_per_diff2  housing_per_diff3  
0             -3.748232          -3.748232  
1361         -22.480211         -22.321900  
2827          -0.210194          -0.210194

In [37]:
# checking how many rows are in each county to make sure the estimates are correct
graham_final["full_fips"].value_counts()

37075920100    1899
37075920300    1466
37075920200    1361
Name: full_fips, dtype: int64

In [38]:
#subsetting by tract
tract1 = graham_final[graham_final["full_fips"] == "37075920100"]
tract2 = graham_final[graham_final["full_fips"] == "37075920200"]
tract3 = graham_final[graham_final["full_fips"] == "37075920300"]

In [39]:
#getting description counts per tract
tract1["PARUSEDESC"].value_counts()

Rural homesite               929
Single family residential    434
Manufactured home            272
Single family high value     263
Single family exceptional      1
Name: PARUSEDESC, dtype: int64

In [40]:
tract2["PARUSEDESC"].value_counts()

Rural homesite               764
Manufactured home            292
Single family residential    256
Single family high value      44
Patio home                     4
Townhouse apartment            1
Name: PARUSEDESC, dtype: int64

In [41]:
tract3["PARUSEDESC"].value_counts()

Rural homesite               905
Manufactured home            317
Single family residential    226
Garden apartment               8
Single family high value       6
Duplex/triplex                 3
Townhouse apartment            1
Name: PARUSEDESC, dtype: int64